src/components/RepositoryGoals.js

```diff
@@ -8,24 +8,29 @@ import AddRepoForm from "../components/AddRepoForm";
 import Cards from "./Card";
 import {done_checking} from "../illustrations";
 import {ContextStyle} from "../styles/Card";
-import api from "../lib/apiGraphQL";
+import {goalsReducer, usePersistentStateReducer} from "../lib/reducers";
 
 function RepositoryGoals() {
-  const [repository, setRepository] = useState({});
-  const [loading, setLoading] = useState(true);
   const {goalsId, setGoalsId} = useContext(LocaleContext);
+  const [state, dispatch] = usePersistentStateReducer("goalsState", goalsReducer);
+
+  const {repository} = goalsReducer(state, {type: "GET"});
+
+  useEffect(() => {
+    repository !== undefined && setGoalsId(repository.id);
```

Comment:
not very readable, I prefer normal if.

Reply:
Good feedback. This is a habit I picked up from trying to conditionally load components, but for the sake of readability I will switch to an IF